api/src/exam-environment/routes/exam-environment.test.ts

```diff
@@ -0,0 +1,601 @@
+          examId: mock.examId
+        };
+
+        const res = await superPost('/exam-environment/exam/generated-exam')
+          .send(body)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        expect(res).toMatchObject({
+          status: 429,
+          body: {
+            code: 'FCC_EINVAL_EXAM_ENVIRONMENT_PREREQUISITES'
+          }
+        });
+
+        await fastifyTestInstance.prisma.envExamAttempt.update({
+          where: {
+            id: recentExamAttempt.id
+          },
+          data: {
+            // Set start time such that exam has expired, but 24 hours - 1s has passed
+            startTimeInMS:
+              Date.now() -
+              (mock.exam.config.totalTimeInMS + (24 * 60 * 60 * 1000 - 1000))
+          }
+        });
+
+        const body2: Static<typeof examEnvironmentPostExamGeneratedExam.body> =
+          {
+            examId: mock.examId
+          };
+
+        const res2 = await superPost('/exam-environment/exam/generated-exam')
+          .send(body2)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        expect(res2).toMatchObject({
+          status: 429,
+          body: {
+            code: 'FCC_EINVAL_EXAM_ENVIRONMENT_PREREQUISITES'
+          }
+        });
+
+        await fastifyTestInstance.prisma.envExamAttempt.update({
+          where: {
+            id: recentExamAttempt.id
+          },
+          data: {
+            // Set start time such that exam has expired, but 24 hours + 1s has passed
+            startTimeInMS:
+              Date.now() -
+              (mock.exam.config.totalTimeInMS + (24 * 60 * 60 * 1000 + 1000))
+          }
+        });
+        // Generate new exam for user to be assigned
+        const newGeneratedExam = structuredClone(mock.generatedExam);
+        newGeneratedExam.id = mock.oid();
+        await fastifyTestInstance.prisma.envGeneratedExam.create({
+          data: newGeneratedExam
+        });
+
+        const body3: Static<typeof examEnvironmentPostExamGeneratedExam.body> =
+          {
+            examId: mock.examId
+          };
+
+        const res3 = await superPost('/exam-environment/exam/generated-exam')
+          .send(body3)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        // Time is greater than 24 hours. So, request should pass, and new exam should be generated
+        expect(res3).toMatchObject({
+          status: 200,
+          body: {
+            data: {
+              examAttempt: {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                id: expect.not.stringMatching(mock.examAttempt.id)
+              }
+            }
+          }
+        });
```

Comment:
This is a big already a big test. I think it makes sense to have the success be one test. If nothing else the test description only talks about errors.

Reply:
Refer to above reply to review comment about reason for test.

Whilst we could split the test up, it would need to then duplicate the 3 requests. So, there would be nicer test names, but the length of the test would not be smaller.

Reply:
We would? I thought everything from line 323 to here was self-contained. i.e. we set the last attempt to be long enough ago that the cooldown is over and then we ask for a new exam.

Reply:
No, you are correct. I should have double-checked it before commenting.